waku/common/rate_limiter_test.go

```diff
@@ -173,3 +174,20 @@ type mockRateLimiterHandler struct {
 
 func (m *mockRateLimiterHandler) ExceedPeerLimit() error { m.exceedPeerLimit++; return nil }
 func (m *mockRateLimiterHandler) ExceedIPLimit() error   { m.exceedIPLimit++; return nil }
+
+type TestWakuPeer struct {
+	peer *p2p.Peer
+}
+
+func (p *TestWakuPeer) EnodeID() enode.ID {
+	return p.peer.ID()
+}
+
+func (p *TestWakuPeer) IP() net.IP {
+	return p.peer.Node().IP()
+}
+
+func (p *TestWakuPeer) ID() []byte {
+	id := p.EnodeID()
+	return id[:]
+}
```

Comment:
This seems to be the only concrete implementation of the new `RateLimiterPeer` interface. Do we not implement `RateLimiterPeer` anywhere else in the code? Or does the `RateLimiterPeer` interface exist only for testing purposes?

Reply:
`Peer` also implements it, before it was actually using the whole `Peer` as an argument, so I initially kept it as it is, but I feel like `RateLimiter` should impose the least amount of requirements on its arguments.

I have actually removed `EnodeID()` as not used (locally, not pushed yet), so basically all `RateLimiter` cares about is something that answer to:

```
func (p *TestWakuPeer) IP() net.IP
func (p *TestWakuPeer) ID() []byte
```

Which intuitively seems to make sense to me, as it is basically a tuple IP/ID, all the info we need to rate limit something.

Reply:
Ah ok, I see that. That's a good implementation.